src/support/rgw_support_bundle

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3
+
+# CORTX Python common library.
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+import sys
+import os
+import shutil
+import tarfile
+import errno
+import subprocess
+import argparse
+
+from cortx.utils.errors import BaseError
+from cortx.utils.process import SimpleProcess
+from cortx.utils.conf_store.conf_store import Conf, MappedConf
+
+
+class SupportBundleError(BaseError):
+    """ SupportBundleError exception with error code and output """
+
+    def __init__(self, rc, message, *args):
+        super().__init__(rc, message, *args)
+
+
+class RGWSupportBundle:
+    """Generate rgw specific support bundle."""
+    _ceph_conf_file = '/etc/ceph/ceph.conf'
+    _tmp_src = '/tmp/rgw/'
+
+    @staticmethod
+    def generate(bundle_id: str, target_path: str, cluster_conf: str, **filters):
+        """ Generate a tar file. """
+        cortx_config_store = MappedConf(cluster_conf)
+        log_base = cortx_config_store.get('cortx>common>storage>log')
+        machine_id = Conf.machine_id
+
+        if os.path.exists(RGWSupportBundle._tmp_src):
+            RGWSupportBundle._cleanup()
+        os.makedirs(RGWSupportBundle._tmp_src, exist_ok=True)
+        # copy configuration files
+        try:
+            shutil.copyfile(RGWSupportBundle._ceph_conf_file,\
+                os.path.join(RGWSupportBundle._tmp_src, 'ceph.conf'))
+        except FileNotFoundError as fe:
+            raise SupportBundleError(errno.ENOENT, "File not found %s", fe)
+
+        # copy rgw client log files
+        log_dir = os.path.join(log_base, f'rgw/{machine_id}')
+        if os.path.exists(log_dir):
+            for file in os.listdir(log_dir):
+                if file.startswith('rgw.client') and file.endswith('.log'):
```

Comment:
please explore on how to collect compressed/rotated log files as well.
we can decide no of log files can be copied based on support bundle size.
e.g. latest 5 files compressed file + 1 live file

Reply:
the script will collect all the log files starting with 'rgw.client' str & ending with '.log'
so, any number of rotated files present will be bundled up.
Later, we can restrict the no. of files if the log file size increases.

Reply:
@gargpalak048 rotated log files would have rgw.client.log.1, .2 etc as extension, which also needs to be taken into consideration

Reply:
Addressed Chetan.
For testing purpose, created dummy log files and tested.
>  tar -xvf rgw_SB_rgw.tar.gz
tar: Removing leading `/' from member names
/
rgw.client.ssc-vm-rhev4-1168.colo.seagate.com.log
ceph.conf
rgw.client.ssc-vm-rhev4-1168.colo.seagate.com-2022-1-03.log
rgw.client.ssc-vm-rhev4-1168.colo.seagate.com.log.1
cortx-rpms
rgw.client.ssc-vm-rhev4-1168.colo.seagate.com.log.2022-01-22

all the format log files are bundled up in tarball